test/ptp/tgm/validation/validation_test.go

```diff
@@ -0,0 +1,105 @@
+//go:build tgmvalidationtests
+// +build tgmvalidationtests
+
+package validation
+
+import (
+	"context"
+	"flag"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/ginkgo/v2/types"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/ptp-operator/test/pkg"
+	"github.com/redhat-eets/sno-tests/test/pkg/client"
+	"github.com/redhat-eets/sno-tests/test/pkg/consts"
+	"github.com/redhat-eets/sno-tests/test/pkg/devices"
+	"github.com/redhat-eets/sno-tests/test/pkg/k8sreporter"
+	"github.com/redhat-eets/sno-tests/test/pkg/namespaces"
+	"github.com/redhat-eets/sno-tests/test/pkg/pods"
+	_ "github.com/redhat-eets/sno-tests/test/ptp/tgm/validation/tests"
+)
+
+var (
+	testClient *client.ClientSet
+	podOnHost  *corev1.Pod
+
+	OperatorNameSpace = pkg.PtpLinuxDaemonNamespace
+	junitPath         *string
+	reportPath        *string
+	r                 *k8sreporter.KubernetesReporter
+)
+
+func init() {
+	if ns := os.Getenv("OO_INSTALL_NAMESPACE"); len(ns) != 0 {
+		OperatorNameSpace = ns
+	}
+
+	junitPath = flag.String("junit", "", "the path for the junit format report")
+	reportPath = flag.String("report", "", "the path of the report file containing details for failed tests")
+}
+
+func TestValidation(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	_, reporterConfig := GinkgoConfiguration()
+
+	if *junitPath != "" {
+		junitFile := path.Join(*junitPath, "tgm_validation_junit.xml")
+		reporterConfig.JUnitReport = junitFile
+	}
+
+	if *reportPath != "" {
+		kubeconfig := os.Getenv("KUBECONFIG")
+		r = k8sreporter.New(kubeconfig, *reportPath, OperatorNameSpace)
```

Comment:
`r.Dump` is only called if the `reportPath` was set, otherwise to which path we print the logs to? We need a path or the print will fail. Even if we set a default path, it may not exist or we might have limited permissions, in that case the test suite will fail.

Reply:
My mistake, all good then!